master/internal/sproto/resource_pool.go

```diff
@@ -0,0 +1,13 @@
+package sproto
+
+type (
+	// CapacityCheck checks the potential available slots in a resource pool.
+	CapacityCheck struct {
+		Slots int
+	}
+	// CapacityCheckResponse is the response to a CapacityCheck message.
+	CapacityCheckResponse struct {
+		SlotsAvailable   int
```

Comment:
The `slotsAvailable` is not being used but I thought it would be great information to have for future use.